builtin/providers/cloudstack/resource_cloudstack_affinity_group.go

```diff
@@ -0,0 +1,136 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackAffinityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackAffinityGroupCreate,
+		Read:   resourceCloudStackAffinityGroupRead,
+		Delete: resourceCloudStackAffinityGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceCloudStackAffinityGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	cs := meta.(*cloudstack.CloudStackClient)
+
+	name := d.Get("name").(string)
+	affinityGroupType := d.Get("type").(string)
+
+	log.Printf("[DEBUG] creating affinity group with name %s of type %s", name, affinityGroupType)
+
+	p := cs.AffinityGroup.NewCreateAffinityGroupParams(name, affinityGroupType)
+
+	// Set the description
+	description, hasDescription := d.GetOk("description")
```

Comment:
It's more idiomatic to write this as: `description, ok := d.GetOk("description"`. So to keep things consistent with the existing code, please also use that here.